app/api/routes.py

```diff
@@ -179,6 +182,57 @@ def get_resources():
     return standardize_response(payload=dict(data=resource_list, **pagination_details))
 
 
+def search(term):
+    query = Resource.query
+    page = 0
+
+    if request.args.get('page') is not None:
+        page = int(request.args.get('page'))
+
+    indicies = client.list_indices()
+    for ind in indicies['items']:
+        if ind['name'] == 'resources_api':
+            curr_index = ind
+
+            if curr_index['entries'] != query.count():
+                print("ReIndexing Database")
+
+                index.clear_objects()
+                db_list = [u.serialize for u in query.all()]
+                index.save_objects(db_list)
+
+            break
+
+    search_results = index.search(f'{term}', {
+        'attributesToRetrieve': [
+            'name',
+            'url',
+            'category',
+            'languages',
+            'notes',
+        ],
+        'page': page,
+    })
+
+    if page >= int(search_results['nbPages']):
+        return redirect('/404')
+
+    # Translate search to normal format.
+    results = []
+    for result in search_results['hits']:
+        results.append(query.get(int(result['objectID'])).serialize)
```

Comment:
Algolia returns a lot of meta data on the search, so here we just get the id of each record and return the serialized object from our db.

Reply:
I feel like there has to be a more efficient way to do this. Let's suppose that algolia returns 80 results. This is going to query the database 80 times to get all of the results from the database? That would take way too long

Reply:
You're right, I should write a join and query once.